File: 04-nested-loops/my-code/202212/lab04.py
def operate_modulo(operand_1: int, operand_2: int) -> None:
	if operand_1 == operand_2:
		print(0)
		return
	for x in range(1, abs(operand_1 - operand_2) + 1):
		if (operand_1 - operand_2) % x == 0:
			print(x, end = " ") if x != abs(operand_1 - operand_2) else print(x)


def operate(operator: str, operand_1: int, operand_2: int) -> bool:
	if operator == "0":
		return False
	match operator:
		case "+":
			print(operand_1 + operand_2)
		case "-":
			print(operand_1 - operand_2)
		case "*":
			print(operand_1 * operand_2)
		case "/":
			print(operand_1 // operand_2, operand_1 % operand_2)
		case ";":
			operate_modulo(operand_1, operand_2)
	return True


def main() -> None:
	resume = True
	while resume:
		operator, operand_1, operand_2 = input().split()
		operand_1 = int(operand_1)
		operand_2 = int(operand_2)
		resume = operate(operator, operand_1, operand_2)


if __name__ == "__main__":
	main()

File: 04-nested-loops/my-code/202212/lab04.c
#include <stdio.h>
#include <stdlib.h>

void operate_modulo(int operand_1, int operand_2) {
	if (operand_1 == operand_2) {
		printf("0\n");
		return;
	}
	for (long int x = 1; x < abs(operand_1 - operand_2) + 1; x++)
		if ((operand_1 - operand_2) % x == 0)
			x != abs(operand_1 - operand_2) ? printf("%ld ", x) : printf("%ld\n", x);
}

int operate(char operator, long int operand_1, long int operand_2) {
	if (operator == '0')
		return 0;
	switch (operator) {
		case '+':
			printf("%ld\n", operand_1 + operand_2);
			break;
		case '-':
			printf("%ld\n", operand_1 - operand_2);
			break;
		case '*':
			printf("%ld\n", operand_1 * operand_2);
			break;
		case '/':
			printf("%ld %ld\n", operand_1 / operand_2, operand_1 % operand_2);
			break;
		case ';':
			operate_modulo(operand_1, operand_2);
			break;
	}
	return 1;
}

int main(void) {
	int resume = 1;
	while (resume) {
		char operator;
		long int operand_1, operand_2;
		scanf(" %c %ld %ld", &operator, &operand_1, &operand_2);
		resume = operate(operator, operand_1, operand_2);
	}
	return 0;
}
